src/flb_encoding.c

```diff
@@ -0,0 +1,159 @@
+/* -*- Mode: C; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- */
+
+/*  Fluent Bit
+ *  ==========
+ *  Copyright (C) 2019      The Fluent Bit Authors
+ *  Copyright (C) 2015-2018 Treasure Data Inc.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+#include <stdio.h>
+
+#include <string.h>
+#include <time.h>
+#include <ctype.h>
+
+#include <fluent-bit/flb_macros.h>
+#include <fluent-bit/flb_config.h>
+#include <fluent-bit/flb_error.h>
+#include <fluent-bit/flb_mem.h>
+#include <fluent-bit/flb_str.h>
+#include <fluent-bit/flb_encoding.h>
+
+
+#include <tutf8e.h>
+
+/*
+ *
+ *  flb_encoding_open(encoding):
+ *  iso-8859-1,...
+ *  windows-1251 windows-1252, ..
+ * 
+ *  <charset>                     - fail if bad chars
+ *  <charset>//IGNORE             - ignore bad chars
+ *  <charset>//REPLACEMENT //R    - use unicode replacement chars for bad chars
+ *  <charset>//QUESTION  //Q      - use '?' for bad chars
+ *  <charset>///<str>             - use <str> for bad chars
+ */
+
+
+static  unsigned char replacement_utf8[] = { 0xEF, 0xBF, 0xBD , 0 };
+
+struct flb_encoding *flb_encoding_open(const char *encoding) {
+    struct flb_encoding *ec;
+    TUTF8encoder encoder;
+    const char *invalid;
+    char *opt;
+
+    if ((opt = strstr(encoding,"//")) != NULL) {
+        *opt = 0;
+        opt += 2;
+        if (*opt == '/') {
+            invalid = opt + 1;
+        }
+        else if (!strcmp(opt,"I") || !strcmp(opt,"IGNORE")) {
+            invalid = "";
+        }
+        else if (!strcmp(opt,"R") || !strcmp(opt,"REPLACEMENT")) {
+            invalid = (const char *) replacement_utf8;
+        }
+        else if (!strcmp(opt,"Q") || !strcmp(opt,"QUESTION")) {
+            invalid = "?";
+        }
+        else {
+            flb_error("[flb_encoding] unknown encodig option: %s", opt);
+            return NULL;
+        }
+    }
+    else {
```

Comment:
Recommend a documentation PR to go along with this over at: 
https://github.com/fluent/fluent-bit-docs

Reply:
Documentation  for change:

https://github.com/fluent/fluent-bit-docs/pull/410